### PR TITLE
Updating Reqwest and some Catalyst Cruft

### PR DESCRIPTION
**Pull request by raayan**
Updating the reqwest version since the old version we were using doesn't let us build for catalyst. Also why use an old version?
Beginnings of getting catalyst to work.

### CODE REVIEW

File: core/Cargo.toml [hunk 1]
```diff
@@ -8,19 +8,22 @@ license = "BSD-3-Clause"
...
+crate-type = ["lib", "staticlib"]
 
 [dependencies]
-reqwest = "0.9.24"
+reqwest = { version = "0.10.4", features = ["blocking", "json"] }
 base64 = "0.11.0"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0.44"
 uuid = { version = "0.8.1", features = ["v4"] }
-sled = { version = "0.31.0" }
```
Comment (Parth) [inline core/Cargo.toml:19]: Will we always need this change?

File: core/Cargo.toml [hunk 2]
```diff
@@ -8,19 +8,22 @@ license = "BSD-3-Clause"
...
+sled = { git = "https://github.com/spacejam/sled", branch = "master" }
 tempfile = { version = "3.1.0" }
 rsa = { version = "^0.2", features = ["serde1"] }
 rand = "0.7"
 sha2 = "0.8.0"
 aead = "0.2.0"
 aes-gcm = "0.5.0"
 termion = "1.5.5"
+
+[profile.release]
```
Comment (Parth) [inline core/Cargo.toml:28]: What do

File: core/Xargo.toml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+std = { features = ["jemalloc"] }
```
Comment (Parth) [inline core/Xargo.toml:1]: What do

File: core/build_for_xcode.sh [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+BUILD_CATALYST=1
+DO_COPY=1
+
+for arg in "$@"
+do
+    case $arg in
+        -f|--fat)
+        BUILD_FAT=1
+        BUILD_CATALYST=0
+        echo "Only building the fatty"
```
Comment (Parth) [inline core/build_for_xcode.sh:13]: Only building fat binary.

File: core/Cargo.toml (hunk 1 above)
Comment (raayan) [reply core/Cargo.toml:19]: No once it's resolved and he bumps the version it goes back, I'll add a TODO

File: core/Xargo.toml (hunk 3 above)
Comment (raayan) [reply core/Xargo.toml:1]: It sets the memory allocator to `jemalloc`

**Pull request merged**